### PR TITLE
Feature/parameterise domain suffix and creds

### PR DESCRIPTION
**Pull request by andywis**
N.B. this branch was created from the make-tests-work-again branch. See  PR#12 (which should be merged first)

### CODE REVIEW

File: setup-ci.sh [hunk 1]
```diff
@@ -43,10 +49,26 @@ function setup_openshift_deployment_jenkins_pipeline() {
 
 function setup_openstack_variables() {
     oc create -f openshift-yaml/openstack_params.yaml
-    oc create secret generic openstack --from-literal=username=openshift@ukcloud.com --from-literal=password=$password
-    oc create secret generic rhelsubscriptions --from-literal=rhel_org=$rhorg --from-literal=rhel_activation_key=$rhactivationkey
-    oc create secret generic s3parameters --from-literal=s3accesskey=$s3accesskey --from-literal=s3secretkey=$s3secretkey --from-literal=s3regionendpoint=$s3regionendpoint --from-literal=s3bucketname=$s3bucketname
-    oc create secret generic openshift --from-literal=OPENSHIFT_USERNAME=$openshiftusername --from-literal=OPENSHIFT_PASSWORD=$openshiftpassword
```
Comment (andywis) [inline setup-ci.sh:49]: this line never worked

File: jenkins-pipelines/openshift/Jenkinsfile [hunk 2]
```diff
@@ -52,26 +54,61 @@ node ('openstack-jenkins-slave') {
...
+        The environment variables set by the pipeline need to be overwritten.
+        The kubernetes parts of the pipeline automatically override the target
+        host, so therefore need to unset them so you can deploy pods within
+        the correct environment.
+
     */
     stage('validate openshift deployment') {
-        sh("git clone -b $openshiftDeploymentAnsibleBranch https://github.com/UKCloud/openshift-deployment-ansible.git openshift-deployment-ansible")
-        sh("for var in \$(export | grep KUB | awk '{ print \$2 }' | sed 's/\\=.*//g'); do unset \$var; done; source ./openstack_rc.sh ; ansible-playbook ./openshift-deployment-ansible/tests/all.yml --extra-vars \"hostIP=$bastionip\" --extra-vars OPENSHIFT_USERNAME=\"\$OPENSHIFT_USERNAME\" --extra-vars OPENSHIFT_PASSWORD=\"\$OPENSHIFT_PASSWORD\" --extra-vars server=\"https://ocp.customer2.cor00005.cna.ukcloud.com:8443\"") }
+        sh("""
```
Comment (andywis) [inline jenkins-pipelines/openshift/Jenkinsfile]: all the password stuff can come out now

File: jenkins-pipelines/openshift/Jenkinsfile [hunk 3]
```diff
@@ -52,26 +54,61 @@ node ('openstack-jenkins-slave') {
...
         sh("scp -o StrictHostKeyChecking=no -i id_rsa_jenkins id_rsa_jenkins cloud-user@$bastionip:")
         sh("ssh -o StrictHostKeyChecking=no -i id_rsa_jenkins cloud-user@$bastionip 'chmod 600 id_rsa_jenkins'")
         sh("ssh -o StrictHostKeyChecking=no -i id_rsa_jenkins cloud-user@$bastionip 'cd /usr/share/ansible/openshift-deployment-ansible ; git checkout $openshiftDeploymentAnsibleBranch'")
-
     }
 
     stage('deploy openshift') {
-        sh("ssh -o StrictHostKeyChecking=no -i id_rsa_jenkins cloud-user@$bastionip 'cd /usr/share/ansible/openshift-deployment-ansible ; ./deploy-openshift.sh'")
+        sh("""
+            ssh -o StrictHostKeyChecking=no -i id_rsa_jenkins cloud-user@$bastionip "cd /usr/share/ansible/openshift-deployment-ansible ; ./deploy-openshift.sh \$OPENSHIFT_PASSWORD"
```
Comment (andywis) [inline jenkins-pipelines/openshift/Jenkinsfile]: need to source openstackrc to pick up this var

**Review by stevemul — APPROVED** [review]
lgtm - will need changes to the pipeline environment that already exists, but otherwise happy with this.

**Pull request merged**